utils.js

```diff
@@ -1,263 +1,295 @@
 'use strict';
 
+const push = require('./config').PUSH;
 const messages = require('./messages');
-const path = require('path');
-const isTar = require('is-tar');
+const fileType = require('file-type');
 
-//Simple function to handle an error occurring in a function that I didn't explicitly catch.
+
+//Get the size of a buffer if a size wasn't given in the file description
+const getStringByteSize = string => Buffer.byteLength(string, 'utf8');
+
+
+//Simple function to handle an error occurring in a function that wasn't explicitly caught.
 const handleError = error => {
   if (typeof error === 'string') {
     throw new Error('Error occurred during processing: ' + error);
   }
-
   throw error;
 };
 
-//This function is to allow for a few more types of archive files to be used instead of just a zip file. Includes tar and the tar extensions .tar.gz or .tar.bz2 (and their short hands).
-//This function takes the contents and transforms them into a zip file buffer, then sends off to the zip file handler to process. It'd be a waste to completely erase the
-//zipHandler method, since it uses JSZip which is very fast and useful. This can be expanded upon in the future, but encompassing all archive file types would be very labor intensive.
-const convertToZip = details => {
 
-  //Check if the tar header is good or exists then proceed
-  if( !isTar(details.content) ){
-    throw new Error(messages.BAD_TAR);
+//Function to handle the logic of all Coveo errors that can occur
+//to make them more user friendly and sorted.
+const coveoErrorHandler = status => {
+  if(status === 400){
+    throw new Error(messages.ERROR_400);
+  } else if (status === 401){
+    throw new Error(messages.ERROR_401);
+  } else if (status === 403){
+    throw new Error(messages.ERROR_403);
+  } else if (status === 412){
+    throw new Error(messages.ERROR_412);
+  } else if (status === 413){
+    throw new Error(messages.ERROR_413);
+  } else if (status === 429){
+    throw new Error(messages.ERROR_429);
+  } else if (status >= 500 || status === 415 || status === 404){
+    throw new Error(messages.SHOULD_NOT_OCCUR_ERRORS);
+  } else {
+    throw new Error(messages.FALLBACK_ERROR);
   }
+};
 
-  let archiveFile = details;
-
-  //This is all for decompressing the contents of the files See: https://www.npmjs.com/package/decompress
-  const decompress = require('decompress'); //Look for alternative for getting all of the contents of all tar file types, as of now there are no good tar readers from a buffer, only local files
-  const zip = require('node-native-zip'); //Consider swapping to this in the future: https://github.com/archiverjs/node-archiver
+//function to check whether or not the url given for the fetch
+//is absolute or the fetch got bad content. 
+const fetchChecker = (url, fetchResponse) => {
+  const lowerCase = require('lower-case');
```

Comment:
You don't need an extra dependency for lowercase.

JavaScript has it on strings: `str.toLowerCase()` and `str.toLocaleLowerCase()`.